### PR TITLE
Jenkins integration

### PR DESCRIPTION
**Pull request by VictorBjelkholm**
Integration with Jenkins, basically just runs the node unit tests for now.

### CODE REVIEW

**Comment by diasdavid** [comment]
Can we have these files inside a 'CI' folder?

**Comment by VictorBjelkholm** [comment]
@diasdavid 
> Can we have these files inside a 'CI' folder?

Sure, good idea! We should do the same across all repositories that is gonna use CI.

**Comment by VictorBjelkholm** [comment]
@diasdavid If this looks fine to you, could you merge it?

**Comment by VictorBjelkholm** [comment]
@diasdavid the work on this PR + https://github.com/ipfs/js-ipfs-api/pull/521 is done at this point. Could you merge both of them?

File: package.json [hunk 1]
```diff
@@ -24,6 +24,8 @@
     "lint": "aegir-lint",
     "coverage": "gulp coverage",
     "test": "gulp test",
+    "test:node": "npm run test:unit:node",
+    "test:browser": "npm run test:unit:browser",
```
Comment (diasdavid) [inline package.json]: ?

**Comment by diasdavid** [comment]
@VictorBjelkholm can I get a debrief? How should I treat jenkins results? Should I block a PR based on Jenkins? Is there any caveat?

**Comment by diasdavid** [comment]
Note, clicking the CI thing leads me to a non valid URL: `http://unconfigured-jenkins-location/job/js-ipfs/job/jenkins-integration/1/display/redirect`

??

File: package.json (hunk 1 above)
Comment (VictorBjelkholm) [inline package.json]: For making it "compatible" with other projects that only exposes "test:node" and "test:browser" as scripts.

**Comment by VictorBjelkholm** [comment]
> @VictorBjelkholm can I get a debrief? How should I treat jenkins results? Should I block a PR based on Jenkins? Is there any caveat?

Basically, right now we're still developing everything Jenkins and setting everything up, so I would say ignore the results for now. However, Jenkins should fail if Circle/Travis fails and same if they succeed, since we're running the same tests. If you see any weird results, it would be good if you report them to me.

> Note, clicking the CI thing leads me to a non valid URL: http://unconfigured-jenkins-location/job/js-ipfs/job/jenkins-integration/1/display/redirect

Yeah, js-ipfs/js-ipfs-api jobs in Jenkins have yet to deployed, hence the URL is no good. As soon as we deploy it to ci.ipfs.team, it'll be all right.

**Comment by diasdavid** [comment]
> Yeah, js-ipfs/js-ipfs-api jobs in Jenkins have yet to deployed, hence the URL is no good. As soon as we deploy it to ci.ipfs.team, it'll be all right.

I'll wait for this. Having a thing that says the tests pass without having a direct path to check what failed will result in confusion.

**Comment by VictorBjelkholm** [comment]
Wasn't there last time you commented but now it is :)

**Pull request merged**